src/nameutil.rs

```diff
@@ -125,6 +125,28 @@ pub fn lib_name_to_toml(name: &str) -> String {
     name.to_string().replace("-", "_").replace(".", "_")
 }
 
+pub fn shared_lib_name_to_link_name(name: &str) -> String {
```

Comment:
```suggestion
pub fn shared_lib_name_to_link_name(name: &str) -> &str {
```

Then you don't need the `to_string()` in the end